pkg/server/sandbox_run.go

```diff
@@ -303,8 +304,62 @@ func (c *criContainerdService) setupSandboxFiles(rootDir string, config *runtime
 	if err := c.os.CopyFile(etcHosts, sandboxEtcHosts, 0666); err != nil {
 		return fmt.Errorf("failed to generate sandbox hosts file %q: %v", sandboxEtcHosts, err)
 	}
-	// TODO(random-liu): [P0] Set DNS options. Maintain a resolv.conf for the sandbox.
+
+	// Set DNS options. Maintain a resolv.conf for the sandbox.
+	var err error
+	resolvContent := ""
+	if dnsConfig := config.GetDnsConfig(); dnsConfig != nil {
+		resolvContent, err = parseDNSOptions(dnsConfig.Servers, dnsConfig.Searches, dnsConfig.Options)
+		if err != nil {
+			return fmt.Errorf("failed to parse sandbox DNSConfig %+v: %v", dnsConfig, err)
+		}
+	}
+	resolvPath := getResolvPath(rootDir)
+	hostNetwork := config.GetLinux().GetSecurityContext().GetNamespaceOptions().GetHostNetwork()
+	if resolvContent == "" && hostNetwork {
+		// copy host's resolv.conf to resolvPath
+		err = c.os.CopyFile(resolvConfPath, resolvPath, 0644)
+		if err != nil {
+			return fmt.Errorf("failed to copy host's resolv.conf to %q: %v", resolvPath, err)
+		}
+	} else {
+		err = c.os.WriteFile(resolvPath, []byte(resolvContent), 0644)
+		if err != nil {
+			return fmt.Errorf("failed to write resolv content to %q: %v", resolvPath, err)
+		}
+	}
+
 	// TODO(random-liu): [P0] Deal with /dev/shm. Use host for HostIpc, and create and mount for
 	// non-HostIpc. What about mqueue?
 	return nil
 }
+
+// parseDNSOptions parse DNS options into resolv.conf format content,
+// if none option is specified, will return empty with no error.
+func parseDNSOptions(servers, searches, options []string) (string, error) {
+	nServers := len(servers)
+	nSearches := len(searches)
+	nOptions := len(options)
+	resolvContent := ""
+	if nServers == 0 && nSearches == 0 && nOptions == 0 {
```

Comment:
nit: This seems unnecessary, the function will return empty string anyway if all are empty.

And if we get rid of this block, we could directly use `len` in the following if clause, because we only need the result once.